src/thor/route_action.cc

```diff
@@ -455,32 +482,29 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
 
       // Build trip path for this leg and add to the result if this
       // location is a BREAK or if this is the last location
-      if (origin->type() == valhalla::Location::kBreak ||
-          origin->type() == valhalla::Location::kBreakThrough) {
+      if (is_break_point(*origin)) {
         // Move destination back to the last break and collect the throughs
         std::list<valhalla::Location> throughs;
-        while (destination->type() != valhalla::Location::kBreak &&
-               destination->type() != valhalla::Location::kBreakThrough) {
+        while (!is_break_point(*destination)) {
           throughs.push_back(*destination);
           --destination;
         }
 
         // We have to flip the via indices because we built them in backwards order
-        decltype(vias) flipped;
+        std::remove_reference<decltype(vias)>::type flipped;
```

Comment:
was `decltype` previously declaring the new variable `flipped` as a reference?

Reply:
it was a compilation error on windows

Reply:
because you captured it by reference when you moved the code into a lambda. i see